src/core/model_repository_manager.cc

```diff
@@ -869,8 +893,8 @@ ModelRepositoryManager::Create(
       tf_allow_soft_placement, tf_memory_limit_mb, &backend_config_map);
 
   std::unique_ptr<BackendLifeCycle> life_cycle;
-  RETURN_IF_ERROR(
-      BackendLifeCycle::Create(server, backend_config_map, &life_cycle));
+  RETURN_IF_ERROR(BackendLifeCycle::Create(
+      server, status_manager.get(), backend_config_map, &life_cycle));
```

Comment:
Why extract the stastus_manager pointer out of the shared_ptr? Since you are capturing it in BackendLifeCycle you should keep it as a shared_ptr. It gets confusing (and dangerous) to have the same pointer be both managed and unmanaged.